emokabe/CapstoneProject
# Dynamic fetch

### Description
This PR implements a dynamic fetching algorithm to get posts from a Facebook Group. The app fetches posts from the most recent two-day timeframe (a relatively short timeframe for testing purposes), ranging from two days ago to the end of today. The algorithm sort through the posts and filters out posts that do not have the correct course id. If there are not enough posts to display, the app fetches a second batch form a two-day timeframe, and keeps fetching posts in batches until either enough posts are displayed or there are no more posts available.

### Milestones
This PR is the start of the "Comment on posts" Stretch feature. On the outside, this looks the same as the "Display the user feed" MVP feature, but implements dynamic fetching to only fetch the most recent posts.

### Test plan
1. Run the simulator, then click the login button to login with Facebook credentials.
2. Once you see the feed, click on the second tab bar button from the left
3. Tap on the "ENGR 40M" course in the list of courses
4. The user should see a feed of posts.
5. Notice that posts that are more than two days apart are shown, even though we are only fetching posts in two-day batches. The recursive strategy for dynamic fetching works.

https://user-images.githubusercontent.com/107252243/181177698-9c4c5722-d5ab-4f1b-ac61-a2c90ef9c590.mov
